### PR TITLE
Read trials from remote storage whenever `get_all_trials` of `_CachedStorage` is called

### PR DESCRIPTION
**Pull request by not522**
## Motivation
Please merge #4670 first.

By #4631, `read_trials_from_remote_storage` is now always called before `get_all_trials`. Therefore, the if-statement for `_CachedStorage` has been eliminated and is now handled within `get_all_trials`.

## Description of the changes
- Make `read_trials_from_remote_storage` a private method.
- Call `_read_trials_from_remote_storage` in `get_all_trials`

### CODE REVIEW

**Comment by not522** [comment]
#4670 has been merged and this PR is ready for review.

**Review by c-bata — APPROVED** [review]
LGTM. I confirmed that this PR does not cause a performance degradation. Here are the benchmark script and the results.

**Benchmark Script**
https://gist.github.com/c-bata/fa8efc075e86a1f7e6dd6d83b85c1017

**Benchmark Result**
| Branch | `n_params=5` (Average Time ± Std Deviation) (s) | `n_params=30` (Average Time ± Std Deviation) (s) |
| --- | --- | --- |
| master `ee15fcc95` | 10.7897 (+/- 0.0556) | 38.0671 (+/- 0.5129) |
| [PR 4672](https://github.com/optuna/optuna/pull/4672) | 10.6553 (+/- 0.1029) | 38.3893 (+/- 1.0469) |

**Comment by toshihikoyanase** [comment]
@gen740 Could you review this PR, please?

**Review by gen740 — APPROVED** [review]
LGTM!

File: optuna/study/study.py [hunk 1]
```diff
@@ -535,8 +529,6 @@ def ask(
 
         # Sync storage once every trial.
         self._thread_local.cached_all_trials = None
-        if isinstance(self._storage, _CachedStorage):
-            self._storage.read_trials_from_remote_storage(self._study_id)
```
Comment (toshihikoyanase) [inline optuna/study/study.py:539]: [Notes] `_pop_waiting_trial_id` internally calls `get_all_trials`. So we can remove this line.
https://github.com/optuna/optuna/blob/f4eec0eea71a8155f1c32f14b7e20f95c061d97b/optuna/study/study.py#L1036-L1038

**Pull request merged**